jqjq.test

```diff
@@ -993,6 +993,16 @@ bsearch(0,2,4)
 1
 -4
 
+# shadowing a passed-through intrinsic
+def type: 42; type
+null
+42
+
+# shadowing a builtin
+def first: 42; first
+null
+42
+
```

Comment:
Nice with test for both! do we want a testing making sure the old function is still callable? something like:
`def _orig_type: type; def type: "42"+_orig_type; 123 | type`